python/tank/templatekey.py

```diff
@@ -47,6 +47,12 @@ def __init__(self,
         self.length = length
         self._last_error = ""
 
+        # check that the key name doesn't contain invalid characters
```

Comment:
I think you would have been way better off having TemplateKey.NAME_REGEX and have template.py use it. It would have been more self-contained.